molgenis/molgenis-emx2
# feat: force dashboard

This PR introduces a new vue app for the FORCE project. The purpose of this vue application is to provide a series of interactive visualisations that can be used to explore the dataset. To do this, a dashboard-like UI will be created where all charts can be clicked, and the clicked value updates the overall data query. 

## To Do

- [x] Setup a new vue application and add to the apps config
- [x] Create the initial file structure
- [x] Create a series of charts based on the data structure
- [x] Make all charts interactive and clicked values are added to a filter
- [x] Create a function that renders the filters into graph ql format
- [x] Integrate the filters into all graphql queries
- [x] Create a component that renders clicked values as filter toggles and link with the filter
- [x] Simplify data retrieval and initial transformations (e.g., selecting columns, sorting rows, transforming for pie charts, etc.)
- [x] Fix rendering issue with column charts when filters are removed
- ~~[  ] Connect chart titles and descriptions to selected filters (TBD)~~
- [x] Fix in BarChart: create a new formula for calculating `y` and `dy` when breaking y-axis ticks
- [x] Add typescript
- [x] Apply styling

**Post-review changes**

- [x] Change button label "Update Charts" to "Apply Filters"
- [x] In the BarChart component, when hovering over a bar, there isn't enough space between the bar and the text. Adjust the `dx` to increase the space

## To Test

**This section is in process. Instructions will be more clear when model is integrated into the catalogue** 

For initial testing and review:

1. Navigate to the preview and login
2. Click on schema `FORCE` to view the dashboard

To use the dashboard:

1. Click on one or more of the chart elements (bars, table rows, slices, etc.) to create a new filter. Filters will appear in the "selected filters" bar. Click "Update Charts" to apply the filters
2. To remove a filter, click the minus icon. (the charts will automatically refresh)
3. Click the "Remove all" button to deselect all filters

Things to check:

- Charts respond to filters and the values are correct
- Chart layout is responsive and content isn't compressed
- Layout does not change (too much) when many filters are applied

## Remaining Questions

- [ ] Are the buttons useful?
- [ ] Should charts update automatically when an element is clicked? Or only when the button is clicked? => Morris says: automatic.
- [ ] What titles should be used here?
- [ ] How can we make the functions more generic? I'm guessing it would be good to have a page for the samples table too
- [ ] Should the standard molgenis filters be used here? I feel that this approach is simpler than the emx2 filters.
